torch/fx/symbolic_trace.py

```diff
@@ -0,0 +1,84 @@
+import inspect
+from types import CodeType, FunctionType
+import torch
+
+from .graph import Graph
+from .graph_module import GraphModule
+
+HAS_VARSTUFF = inspect.CO_VARARGS | inspect.CO_VARKEYWORDS
+
+def _find_module(root, m):
+    for n, p in root.named_modules():
+        if m is p:
+            return n
+    raise NameError('module is not installed as a submodule')
+
+def _patch_function(fn, nargs):
+    co = fn.__code__
+    co_flags = co.co_flags & ~HAS_VARSTUFF
+    if hasattr(co, "co_posonlyargcount"):
+        co_args = (
+            nargs, 0,
+            0, co.co_nlocals, co.co_stacksize,
+            co_flags, co.co_code, co.co_consts, co.co_names,
+            co.co_varnames, co.co_filename, co.co_name,
+            co.co_firstlineno, co.co_lnotab, co.co_freevars,
+            co.co_cellvars
+        )
+    else:
+        co_args = (
+            nargs, 0, co.co_nlocals,
+            co.co_stacksize, co_flags, co.co_code, co.co_consts,
+            co.co_names, co.co_varnames, co.co_filename,
+            co.co_name, co.co_firstlineno, co.co_lnotab,
+            co.co_freevars, co.co_cellvars)
+    new_code = CodeType(*co_args)
+    return FunctionType(new_code, fn.__globals__, fn.__name__, fn.__defaults__, fn.__closure__)
+
+    # we need to insert placeholder nodes for *args, and **kwargs,
+    # so we can't call this function normally, otherwise it would try to unpack them
+    # instead, let's make python think that args and kwargs are normay variables
+
+def is_leaf_module(m):
```

Comment:
How do we handle those cases when there is a control flow in the Module and we'd still like to delegate the execution to backend (given it supports that)?

Reply:
Those should be encapsulated in functional blocks, e.g. an `LSTM` root module. From the perspective of this IR, this will look like a purely functional block, but internally it may contain control flow, which can be lowered to an accelerator natively if supported